README.rst

```diff
@@ -16,8 +16,6 @@ You want to benchmark Elasticsearch? Then Rally is for you. It can help you with
 
 We have also put considerable effort in Rally to ensure that benchmarking data are reproducible.
 
-In general, Rally works with all versions of Elasticsearch starting from 1.x but benchmarking with plugins and benchmarking source builds will only work from Elasticsearch 5.0 onwards.
```

Comment:
Do we want to specify 6.x onwards instead of totally removing it?

Reply:
Our README file mostly exists so people that open the Rally repo on Github have an entry point. This file closely resembles our index page at https://esrally.readthedocs.io/en/stable/ which also does not document this so I opted for consistency across the two pages.